pandas/io/packers.py

```diff
@@ -350,8 +350,11 @@ def unconvert(values, dtype, compress=None):
                 )
                 # fall through to copying `np.fromstring`
 
-    # Copy the string into a numpy array.
-    return np.fromstring(values, dtype=dtype)
+    # Copy the bytes into a numpy array.
+    buf = np.frombuffer(values, dtype=dtype)
+    buf = buf.copy()  # required to not mutate the original data
```

Comment:
I _think_ it's safe to not copy here, IIRC `values` is pile of bytes read from a msgpack file, so no risk in mutating?

Reply:
Without the copy the test failed:

```
        for buf, control_buf in zip(not_garbage, control):
            # make sure none of our mutations above affected the
            # original buffers
>           assert buf == control_buf
E           AssertionError: assert b'\x00\x00\x0...x00\x00@\x8f@' == bytearray(b'\x...00\x008\x8f@')
E             At index 6 diff: 240 != 0
E             Use -v to get the full diff

pandas/tests/io/test_packers.py:690: AssertionError
```

I didn't look closely to see if that's a legitimate problem or not.

Reply:
Ah, in that case I would trust the test.

Reply:
you need to do this conditionally for numpy > 1.13 I think

Reply:
cc @llllllllll

Reply:
The tests on the older NumPy passed, let me make sure zlib is installed and that they weren't skipped.

Reply:
Pretty sure that https://travis-ci.org/pandas-dev/pandas/jobs/292845756 hit it, so I think we're OK.